CONTRIBUTING.md

```diff
@@ -156,6 +156,28 @@ Suppose your plugin is ready for version 1.2.0
 
 ```
 
+### Test Development
+
+#### Experimental Features
+
+When writing tests for experimental features, please follow these steps.
+
+1. figure out the folder location to put the tests
+
+Similar to the regular tests, OSD Core tests go to the [folder](integration/core-opensearch-dashboards/opensearch-dashboards/) and OSD plugin tests go to the [folder](cypress/integration/plugins/).
+
+2. develop tests with a flag to turn on and off
+
+Add an environment variable (e.g boolean) to only run tests for the experiemental feature when it is true. This is to ensure backward compatibility when integrating with [opensearch-build repo](https://github.com/opensearch-project/opensearch-build/blob/main/src/test_workflow/integ_test/service_opensearch_dashboards.py) whose OpenSearch Dashboards execution command or yml configuration may not be updated to support the experimental feature yet.
```

Comment:
nit: Link to the env would be useful i.e. https://github.com/opensearch-project/opensearch-dashboards-functional-test/blob/main/cypress.json#L12

Reply:
updated.